impl/src/main/java/org/jboss/weld/event/ObserverNotifier.java

```diff
@@ -332,6 +334,8 @@ public RuntimeException apply(Type eventType) {
         final NotificationMode mode = initModeOption(options.get(WeldNotificationOptions.MODE));
         final Long timeout = initTimeoutOption(options.get(WeldNotificationOptions.TIMEOUT));
         final Consumer<Runnable> securityContextActionConsumer = securityServices.getSecurityContextAssociator();
+        // grab current TCCL
+        ClassLoader tccl = GetContextClassLoaderAction.INSTANCE.run();
```

Comment:
You can't use a `PrivilegedAction` like this. You should pass the instance as a param of an `AccessController.doPrivileged()` invocation. You could also create a package-private `SecurityActions` util class like https://github.com/weld/core/blob/master/impl/src/main/java/org/jboss/weld/bean/proxy/SecurityActions.java and optimize the code if SM is not enabled.